### PR TITLE
user_types and datetimes

### PR DESCRIPTION
**Pull request by NeilMenne**
`to_kind!` and `to_type!` all suffer behind the curtain of `to_spec!` to make user_types work, but it's minimally invasive otherwise.

### CODE REVIEW

**Comment by NeilMenne** [comment]
[spect: handle other things](https://trello.com/c/EGvTVN0a/78-spect-handle-other-things)

File: lib/spect.ex [hunk 1]
```diff
@@ -153,8 +163,18 @@ defmodule Spect do
       )
   end
 
+  defp to_type!(data, module, :module, _args) do
```
Comment (brentspell) [inline lib/spect.ex:166]: This function isn't covered by the tests. Run `MIX_ENV=test mix coveralls.html` for code coverage.

File: lib/spect.ex [hunk 2]
```diff
@@ -271,32 +291,42 @@ defmodule Spect do
...
   # map
-  defp to_type!(data, :map, args) do
-    to_map!(data, args)
+  defp to_type!(data, module, :map, args) do
+    to_map!(data, module, args)
+  end
+
+  # catch all
+  defp to_type!(data, _module, type, args) do
+    raise(
```
Comment (brentspell) [inline lib/spect.ex]: Looks like this is also not being exercised by the tests.

**Comment by brentspell** [comment]
Other than test coverage, looks good to me.

**Pull request merged**